common/cuda_hip/solver/idr_kernels.cpp

```diff
@@ -555,7 +555,8 @@ void initialize(std::shared_ptr<const DefaultExecutor> exec,
     orthonormalize_subspace_vectors(exec, subspace_vectors);
 }
 
-GKO_INSTANTIATE_FOR_EACH_VALUE_TYPE(GKO_DECLARE_IDR_INITIALIZE_KERNEL);
+GKO_INSTANTIATE_FOR_EACH_VALUE_TYPE_WITH_HALF(
```

Comment:
A general question: Why do we need another macro `_WITH_HALF` ? Is it not possible to just have `half` within the first macro ?

Reply:
It is possible but it will need to result it in #1257  because we need to finish everything all at once.